Quieter `lock` output for large monorepos

The following is the output of a successful `opam monorepo lock` on the Tezos repository:

```
❭ opam monorepo lock --recurse-opam
opam-monorepo: [ERROR] /home/craigfe/t/tezos/_opam: No such file or directory
==> Using locally scanned packages 'benchmark-utils, bls12-381,
bls12-381-gen, bls12-381-unix, flextesa, hacl-star-raw-empty, latex,
ledgerwallet, ledgerwallet-tezos, lib-test, numerics, pyml-plot, staTz,
tezos-008-PtEdo2Zk-test-helpers, tezos-009-PsFLoren-test-helpers,
tezos-010-PtGRANAD-test-helpers, tezos-accuser-008-PtEdo2Zk,
tezos-accuser-008-PtEdo2Zk-commands, tezos-accuser-009-PsFLoren,
tezos-accuser-009-PsFLoren-commands, tezos-accuser-010-PtGRANAD,
tezos-accuser-010-PtGRANAD-commands, tezos-accuser-alpha,
tezos-accuser-alpha-commands, tezos-alpha-test-helpers,
tezos-baker-008-PtEdo2Zk, tezos-baker-009-PsFLoren, tezos-baker-010-PtGRANAD,
tezos-baker-alpha, tezos-baking-008-PtEdo2Zk,
tezos-baking-008-PtEdo2Zk-commands, tezos-baking-009-PsFLoren,
tezos-baking-009-PsFLoren-commands, tezos-baking-010-PtGRANAD,
tezos-baking-010-PtGRANAD-commands, tezos-baking-alpha,
tezos-baking-alpha-commands, tezos-base, tezos-benchmark,
tezos-benchmark-010-PtGRANAD, tezos-benchmark-alpha,
tezos-benchmark-examples, tezos-benchmark-tests,
tezos-benchmark-type-inference-010-PtGRANAD,
tezos-benchmark-type-inference-alpha, tezos-benchmarks-proto-010-PtGRANAD,
tezos-benchmarks-proto-alpha, tezos-clic, tezos-client,
tezos-client-000-Ps9mPmXa, tezos-client-001-PtCJ7pwo,
tezos-client-001-PtCJ7pwo-commands, tezos-client-002-PsYLVpVv,
tezos-client-002-PsYLVpVv-commands, tezos-client-003-PsddFKi3,
tezos-client-003-PsddFKi3-commands, tezos-client-004-Pt24m4xi,
tezos-client-004-Pt24m4xi-commands, tezos-client-005-PsBabyM1,
tezos-client-005-PsBabyM1-commands, tezos-client-006-PsCARTHA,
tezos-client-006-PsCARTHA-commands, tezos-client-007-PsDELPH1,
tezos-client-007-PsDELPH1-commands,
tezos-client-007-PsDELPH1-commands-registration, tezos-client-008-PtEdo2Zk,
tezos-client-008-PtEdo2Zk-commands,
tezos-client-008-PtEdo2Zk-commands-registration, tezos-client-009-PsFLoren,
tezos-client-009-PsFLoren-commands,
tezos-client-009-PsFLoren-commands-registration, tezos-client-010-PtGRANAD,
tezos-client-010-PtGRANAD-commands,
tezos-client-010-PtGRANAD-commands-registration, tezos-client-alpha,
tezos-client-alpha-commands, tezos-client-alpha-commands-registration,
tezos-client-base, tezos-client-base-unix, tezos-client-commands,
tezos-client-demo-counter, tezos-client-genesis,
tezos-client-genesis-carthagenet, tezos-client-sapling-008-PtEdo2Zk,
tezos-client-sapling-009-PsFLoren, tezos-client-sapling-010-PtGRANAD,
tezos-client-sapling-alpha, tezos-codec, tezos-context, tezos-crypto,
tezos-embedded-protocol-000-Ps9mPmXa, tezos-embedded-protocol-001-PtCJ7pwo,
tezos-embedded-protocol-002-PsYLVpVv, tezos-embedded-protocol-003-PsddFKi3,
tezos-embedded-protocol-004-Pt24m4xi, tezos-embedded-protocol-005-PsBABY5H,
tezos-embedded-protocol-005-PsBabyM1, tezos-embedded-protocol-006-PsCARTHA,
tezos-embedded-protocol-007-PsDELPH1, tezos-embedded-protocol-008-PtEdo2Zk,
tezos-embedded-protocol-008-PtEdoTez, tezos-embedded-protocol-009-PsFLoren,
tezos-embedded-protocol-010-PtGRANAD, tezos-embedded-protocol-alpha,
tezos-embedded-protocol-demo-counter, tezos-embedded-protocol-demo-noops,
tezos-embedded-protocol-genesis, tezos-embedded-protocol-genesis-carthagenet,
tezos-endorser-008-PtEdo2Zk, tezos-endorser-008-PtEdo2Zk-commands,
tezos-endorser-009-PsFLoren, tezos-endorser-009-PsFLoren-commands,
tezos-endorser-010-PtGRANAD, tezos-endorser-010-PtGRANAD-commands,
tezos-endorser-alpha, tezos-endorser-alpha-commands, tezos-error-monad,
tezos-event-logging, tezos-hacl-glue, tezos-hacl-glue-js,
tezos-hacl-glue-unix, tezos-legacy-store, tezos-lmdb,
tezos-lwt-result-stdlib, tezos-micheline, tezos-micheline-rewriting,
tezos-mockup, tezos-mockup-commands, tezos-mockup-proxy,
tezos-mockup-registration, tezos-node, tezos-p2p, tezos-p2p-services,
tezos-protocol-000-Ps9mPmXa, tezos-protocol-001-PtCJ7pwo,
tezos-protocol-002-PsYLVpVv, tezos-protocol-003-PsddFKi3,
tezos-protocol-004-Pt24m4xi, tezos-protocol-005-PsBABY5H,
tezos-protocol-005-PsBabyM1, tezos-protocol-006-PsCARTHA,
tezos-protocol-006-PsCARTHA-parameters, tezos-protocol-007-PsDELPH1,
tezos-protocol-008-PtEdo2Zk, tezos-protocol-008-PtEdo2Zk-parameters,
tezos-protocol-008-PtEdo2Zk-tests, tezos-protocol-008-PtEdoTez,
tezos-protocol-009-PsFLoren, tezos-protocol-009-PsFLoren-parameters,
tezos-protocol-009-PsFLoren-tests, tezos-protocol-010-PtGRANAD,
tezos-protocol-010-PtGRANAD-parameters, tezos-protocol-010-PtGRANAD-tests,
tezos-protocol-alpha, tezos-protocol-alpha-parameters,
tezos-protocol-alpha-tests, tezos-protocol-compiler,
tezos-protocol-demo-counter, tezos-protocol-demo-noops,
tezos-protocol-environment, tezos-protocol-environment-packer,
tezos-protocol-environment-sigs, tezos-protocol-environment-structs,
tezos-protocol-functor-000-Ps9mPmXa, tezos-protocol-functor-001-PtCJ7pwo,
tezos-protocol-functor-002-PsYLVpVv, tezos-protocol-functor-003-PsddFKi3,
tezos-protocol-functor-004-Pt24m4xi, tezos-protocol-functor-005-PsBABY5H,
tezos-protocol-functor-005-PsBabyM1, tezos-protocol-functor-006-PsCARTHA,
tezos-protocol-functor-007-PsDELPH1, tezos-protocol-functor-008-PtEdo2Zk,
tezos-protocol-functor-008-PtEdoTez, tezos-protocol-functor-009-PsFLoren,
tezos-protocol-functor-010-PtGRANAD, tezos-protocol-functor-alpha,
tezos-protocol-functor-demo-counter, tezos-protocol-functor-demo-noops,
tezos-protocol-functor-genesis, tezos-protocol-functor-genesis-carthagenet,
tezos-protocol-genesis, tezos-protocol-genesis-carthagenet,
tezos-protocol-plugin-007-PsDELPH1,
tezos-protocol-plugin-007-PsDELPH1-registerer,
tezos-protocol-plugin-008-PtEdo2Zk,
tezos-protocol-plugin-008-PtEdo2Zk-registerer,
tezos-protocol-plugin-009-PsFLoren,
tezos-protocol-plugin-009-PsFLoren-registerer,
tezos-protocol-plugin-010-PtGRANAD,
tezos-protocol-plugin-010-PtGRANAD-registerer, tezos-protocol-plugin-alpha,
tezos-protocol-plugin-alpha-registerer, tezos-protocol-updater, tezos-proxy,
tezos-requester, tezos-rpc, tezos-rpc-http, tezos-rpc-http-client,
tezos-rpc-http-client-unix, tezos-rpc-http-server, tezos-rust-libs,
tezos-sapling, tezos-shell, tezos-shell-benchmarks, tezos-shell-context,
tezos-shell-services, tezos-signer, tezos-signer-backends,
tezos-signer-services, tezos-snoop, tezos-stdlib, tezos-stdlib-unix,
tezos-store, tezos-test-services, tezos-tooling, tezos-validation,
tezos-validator, tezos-version, tezos-workers, tezt, tezt-tezos,
zip' as the roots.
opam-monorepo: [WARNING] The dune-universe opam-repository isn't set in the current switch. It contains dune ports for some opam packages. Note that opam monorepo lock will fail if not all of the project dependencies use dune as their build system. Adding this opam-repository to your current switch will help with that. If you wish to do so, run the following command:
opam repository add dune-universe git+https://github.com/dune-universe/opam-overlays.git
==> Found 166 opam dependencies for benchmark-utils bls12-381 bls12-381-gen bls12-381-unix flextesa hacl-star-raw-empty latex ledgerwallet ledgerwallet-tezos lib-test numerics pyml-plot staTz tezos-008-PtEdo2Zk-test-helpers tezos-009-PsFLoren-test-helpers tezos-010-PtGRANAD-test-helpers tezos-accuser-008-PtEdo2Zk tezos-accuser-008-PtEdo2Zk-commands tezos-accuser-009-PsFLoren tezos-accuser-009-PsFLoren-commands tezos-accuser-010-PtGRANAD tezos-accuser-010-PtGRANAD-commands tezos-accuser-alpha tezos-accuser-alpha-commands tezos-alpha-test-helpers tezos-baker-008-PtEdo2Zk tezos-baker-009-PsFLoren tezos-baker-010-PtGRANAD tezos-baker-alpha tezos-baking-008-PtEdo2Zk tezos-baking-008-PtEdo2Zk-commands tezos-baking-009-PsFLoren tezos-baking-009-PsFLoren-commands tezos-baking-010-PtGRANAD tezos-baking-010-PtGRANAD-commands tezos-baking-alpha tezos-baking-alpha-commands tezos-base tezos-benchmark tezos-benchmark-010-PtGRANAD tezos-benchmark-alpha tezos-benchmark-examples tezos-benchmark-tests tezos-benchmark-type-inference-010-PtGRANAD tezos-benchmark-type-inference-alpha tezos-benchmarks-proto-010-PtGRANAD tezos-benchmarks-proto-alpha tezos-clic tezos-client tezos-client-000-Ps9mPmXa tezos-client-001-PtCJ7pwo tezos-client-001-PtCJ7pwo-commands tezos-client-002-PsYLVpVv tezos-client-002-PsYLVpVv-commands tezos-client-003-PsddFKi3 tezos-client-003-PsddFKi3-commands tezos-client-004-Pt24m4xi tezos-client-004-Pt24m4xi-commands tezos-client-005-PsBabyM1 tezos-client-005-PsBabyM1-commands tezos-client-006-PsCARTHA tezos-client-006-PsCARTHA-commands tezos-client-007-PsDELPH1 tezos-client-007-PsDELPH1-commands tezos-client-007-PsDELPH1-commands-registration tezos-client-008-PtEdo2Zk tezos-client-008-PtEdo2Zk-commands tezos-client-008-PtEdo2Zk-commands-registration tezos-client-009-PsFLoren tezos-client-009-PsFLoren-commands tezos-client-009-PsFLoren-commands-registration tezos-client-010-PtGRANAD tezos-client-010-PtGRANAD-commands tezos-client-010-PtGRANAD-commands-registration tezos-client-alpha tezos-client-alpha-commands tezos-client-alpha-commands-registration tezos-client-base tezos-client-base-unix tezos-client-commands tezos-client-demo-counter tezos-client-genesis tezos-client-genesis-carthagenet tezos-client-sapling-008-PtEdo2Zk tezos-client-sapling-009-PsFLoren tezos-client-sapling-010-PtGRANAD tezos-client-sapling-alpha tezos-codec tezos-context tezos-crypto tezos-embedded-protocol-000-Ps9mPmXa tezos-embedded-protocol-001-PtCJ7pwo tezos-embedded-protocol-002-PsYLVpVv tezos-embedded-protocol-003-PsddFKi3 tezos-embedded-protocol-004-Pt24m4xi tezos-embedded-protocol-005-PsBABY5H tezos-embedded-protocol-005-PsBabyM1 tezos-embedded-protocol-006-PsCARTHA tezos-embedded-protocol-007-PsDELPH1 tezos-embedded-protocol-008-PtEdo2Zk tezos-embedded-protocol-008-PtEdoTez tezos-embedded-protocol-009-PsFLoren tezos-embedded-protocol-010-PtGRANAD tezos-embedded-protocol-alpha tezos-embedded-protocol-demo-counter tezos-embedded-protocol-demo-noops tezos-embedded-protocol-genesis tezos-embedded-protocol-genesis-carthagenet tezos-endorser-008-PtEdo2Zk tezos-endorser-008-PtEdo2Zk-commands tezos-endorser-009-PsFLoren tezos-endorser-009-PsFLoren-commands tezos-endorser-010-PtGRANAD tezos-endorser-010-PtGRANAD-commands tezos-endorser-alpha tezos-endorser-alpha-commands tezos-error-monad tezos-event-logging tezos-hacl-glue tezos-hacl-glue-js tezos-hacl-glue-unix tezos-legacy-store tezos-lmdb tezos-lwt-result-stdlib tezos-micheline tezos-micheline-rewriting tezos-mockup tezos-mockup-commands tezos-mockup-proxy tezos-mockup-registration tezos-node tezos-p2p tezos-p2p-services tezos-protocol-000-Ps9mPmXa tezos-protocol-001-PtCJ7pwo tezos-protocol-002-PsYLVpVv tezos-protocol-003-PsddFKi3 tezos-protocol-004-Pt24m4xi tezos-protocol-005-PsBABY5H tezos-protocol-005-PsBabyM1 tezos-protocol-006-PsCARTHA tezos-protocol-006-PsCARTHA-parameters tezos-protocol-007-PsDELPH1 tezos-protocol-008-PtEdo2Zk tezos-protocol-008-PtEdo2Zk-parameters tezos-protocol-008-PtEdo2Zk-tests tezos-protocol-008-PtEdoTez tezos-protocol-009-PsFLoren tezos-protocol-009-PsFLoren-parameters tezos-protocol-009-PsFLoren-tests tezos-protocol-010-PtGRANAD tezos-protocol-010-PtGRANAD-parameters tezos-protocol-010-PtGRANAD-tests tezos-protocol-alpha tezos-protocol-alpha-parameters tezos-protocol-alpha-tests tezos-protocol-compiler tezos-protocol-demo-counter tezos-protocol-demo-noops tezos-protocol-environment tezos-protocol-environment-packer tezos-protocol-environment-sigs tezos-protocol-environment-structs tezos-protocol-functor-000-Ps9mPmXa tezos-protocol-functor-001-PtCJ7pwo tezos-protocol-functor-002-PsYLVpVv tezos-protocol-functor-003-PsddFKi3 tezos-protocol-functor-004-Pt24m4xi tezos-protocol-functor-005-PsBABY5H tezos-protocol-functor-005-PsBabyM1 tezos-protocol-functor-006-PsCARTHA tezos-protocol-functor-007-PsDELPH1 tezos-protocol-functor-008-PtEdo2Zk tezos-protocol-functor-008-PtEdoTez tezos-protocol-functor-009-PsFLoren tezos-protocol-functor-010-PtGRANAD tezos-protocol-functor-alpha tezos-protocol-functor-demo-counter tezos-protocol-functor-demo-noops tezos-protocol-functor-genesis tezos-protocol-functor-genesis-carthagenet tezos-protocol-genesis tezos-protocol-genesis-carthagenet tezos-protocol-plugin-007-PsDELPH1 tezos-protocol-plugin-007-PsDELPH1-registerer tezos-protocol-plugin-008-PtEdo2Zk tezos-protocol-plugin-008-PtEdo2Zk-registerer tezos-protocol-plugin-009-PsFLoren tezos-protocol-plugin-009-PsFLoren-registerer tezos-protocol-plugin-010-PtGRANAD tezos-protocol-plugin-010-PtGRANAD-registerer tezos-protocol-plugin-alpha tezos-protocol-plugin-alpha-registerer tezos-protocol-updater tezos-proxy tezos-requester tezos-rpc tezos-rpc-http tezos-rpc-http-client tezos-rpc-http-client-unix tezos-rpc-http-server tezos-rust-libs tezos-sapling tezos-shell tezos-shell-benchmarks tezos-shell-context tezos-shell-services tezos-signer tezos-signer-backends tezos-signer-services tezos-snoop tezos-stdlib tezos-stdlib-unix tezos-store tezos-test-services tezos-tooling tezos-validation tezos-validator tezos-version tezos-workers tezt tezt-tezos zip.
==> Querying opam database for their metadata and Dune compatibility.
==> Calculating exact pins for each of them.
==> Wrote lockfile with 107 entries to /home/craigfe/t/tezos/tezos.opam.locked. You can now run opam monorepo pull to fetch their sources.
```

There are two places where the full set of packages is printed, which adds a lot of noise to the terminal output. It seems sensible to just print the number of such packages above some limit (say, 10) unless `--verbose` is passed.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -12,6 +12,8 @@
 - Recognize packages with an optional dependency on dune as building with dune. This allows
   opam-monorepo to rightfully recognize `opam-file-format` latest versions as building with 
   dune. (#176, @NathanReb)
+- Only print the full list of selected root packages once and only in verbose mode, simply printing
+  the number in the default logs. (#173, @NathanReb)
 
 ### Deprecated
 

--- a/cli/lock.ml
+++ b/cli/lock.ml
@@ -2,18 +2,22 @@ open Import
 open Types
 
 let check_root_packages ~local_packages =
-  match local_packages with
-  | [] ->
+  let count = List.length local_packages in
+  match count with
+  | 0 ->
       Rresult.R.error_msg
         "Cannot find any packages to vendor.\n\
          Either create some *.opam files in the local repository, or specify them manually via \
          'opam monorepo lock <packages>'."
-  | local_packages ->
+  | _ ->
       let pp_package_name fmt { Opam.name; _ } = Fmt.string fmt name in
       Common.Logs.app (fun l ->
-          l "Using locally scanned package%a '%a' as the root%a." Pp.plural local_packages
+          l "Using %d locally scanned package%a as the root%a." count Pp.plural local_packages
+            Pp.plural local_packages);
+      Logs.info (fun l ->
+          l "Root package%a: %a." Pp.plural local_packages
             Fmt.(list ~sep:(unit ",@ ") (styled `Yellow pp_package_name))
-            local_packages Pp.plural local_packages);
+            local_packages);
       Ok ()
 
 let opam_to_git_remote remote =

--- a/lib/opam_solve.ml
+++ b/lib/opam_solve.ml
@@ -110,17 +110,11 @@ let calculate ~build_only ~allow_jbuilder ~local_opam_files ~local_packages ?oca
     switch_state
   >>= fun deps ->
   Logs.app (fun l ->
-      l "%aFound %a opam dependencies for %a." Pp.Styled.header ()
+      l "%aFound %a opam dependencies for the root package%a." Pp.Styled.header ()
         Fmt.(styled `Green int)
-        (List.length deps)
-        Fmt.(list ~sep:(unit " ") Pp.Styled.package)
-        local_packages);
+        (List.length deps) Pp.plural local_packages);
   Logs.info (fun l ->
-      l "The dependencies for %a are: %a"
-        Fmt.(list ~sep:(unit ",@ ") Types.Opam.pp_package)
-        local_packages
-        Fmt.(list ~sep:(unit ",@ ") Opam.Pp.package)
-        deps);
+      l "The dependencies are: %a" Fmt.(list ~sep:(unit ",@ ") Opam.Pp.package) deps);
   Logs.app (fun l ->
       l "%aQuerying opam database for their metadata and Dune compatibility." Pp.Styled.header ());
   Ok (List.map ~f:(get_opam_info ~switch_state) deps)

